genref/output/md/cloud-provider-service.v1alpha1.md

```diff
@@ -9,32 +9,29 @@
 ## `ServiceControllerConfiguration`     {#ServiceControllerConfiguration}
     
 
-
-
 **Appears in:**
 
 - [CloudControllerManagerConfiguration](#cloudcontrollermanager-config-k8s-io-v1alpha1-CloudControllerManagerConfiguration)
 
 - [KubeControllerManagerConfiguration](#kubecontrollermanager-config-k8s-io-v1alpha1-KubeControllerManagerConfiguration)
 
 
-ServiceControllerConfiguration contains elements describing ServiceController.
+<p>ServiceControllerConfiguration contains elements describing ServiceController.</p>
+
 
 <table class="table">
 <thead><tr><th width="30%">Field</th><th>Description</th></tr></thead>
 <tbody>
     
-
   
 <tr><td><code>ConcurrentServiceSyncs</code> <B>[Required]</B><br/>
 <code>int32</code>
 </td>
 <td>
-   concurrentServiceSyncs is the number of services that are
+   <p>concurrentServiceSyncs is the number of services that are
```

Comment:
nit: Just checking if the field name is `concurrentServiceSyncs` or `ConcurrentServiceSyncs`?

Reply:
This is a good question. I have to check the upstream source code to give you an answer ...

The source code corresponding to the field is located [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/service/config/v1alpha1/types.go#L19-L25). As you can see from the snippet copied below, the struct definition failed to include a JSON tag for the member field `ConcurrentServiceSyncs`.

```go
// ServiceControllerConfiguration contains elements describing ServiceController.
type ServiceControllerConfiguration struct {
	// concurrentServiceSyncs is the number of services that are
	// allowed to sync concurrently. Larger number = more responsive service
	// management, but more CPU (and network) load.
	ConcurrentServiceSyncs int32
}
```

In other words, the declaration of the field should be something like (**note how the user facing field name is specified using this tag**):

```go
     ConcurrentServiceSyncs int32  `json: "concurrentServiceSyncs"`
```

Without this tag, the configuration will have to contain "`ConcurrentServiceSyncs`" for it to be valid. However, the comment string is still putting it as 'concurrentServiceSyncs'. So this smells like bug in the source code.

Reply:
👍